Std/Lean/Meta/DiscrTree.lean

```diff
@@ -101,6 +107,57 @@ where
       (ord := ⟨compareOn (·.fst)⟩) cs₁ cs₂
       (fun (k₁, t₁) (_, t₂) => (k₁, mergePreservingDuplicates t₁ t₂))
 
+/-- The explicit stack of `mapArrays` -/
+private inductive Ctxt {α β s}
+  | empty : Ctxt
+  | ctxt : Array (Key s × Trie β s) → Array β → Array (Key s × Trie α s) → Key s → Ctxt → Ctxt
+
+/-- Apply a function to the array of values at each node in a `DiscrTree`. -/
+partial def mapArrays (t : Trie α s) (f : Array α → Array β) : Trie β s :=
+  let .node vs0 cs0 := t
+  go (.mkEmpty cs0.size) (f vs0) cs0.reverse Ctxt.empty
+where
+  /-- This implementation as a single tail-recursive function is chosen to not blow the
+      interpreter stack when the `Trie` is very deep -/
```

Comment:
This is not a doc comment. This should be a plain comment and the doc comment should say what the function does. In this case I would try to write a theorem which defines `go` non-recursively in terms of `mapArrays` as a means of specification (see the doc comments on `List.setTR.go` and other worker-wrapper functions in Std.Data.List.Basic for examples).

Reply:
It was a plain comment before :-)
I only changed this because the doc linter insisted one one, despite this being a local definition. Is it a bug in the linter that it insists on docstrings on `where`'d definitions?

Reply:
I’d rather not invest much into this code (e.g. theorems relating to other specifications) if I get to dump it anyways soon if https://github.com/leanprover/lean4/pull/2577 can land. But I addressed the docstring issue.